lib/ex_oauth2_provider/oauth2/authorization/strategy/device_code/device_authorization.ex

```diff
@@ -0,0 +1,111 @@
+defmodule ExOauth2Provider.Authorization.DeviceCode.DeviceAuthorization do
+  alias ExOauth2Provider.{
+    Config,
+    DeviceGrants,
+    Authorization.Utils,
+    Scopes,
+    Utils.Error
+  }
+
+  # Device Authorization Request
+  # https://tools.ietf.org/html/rfc8628#section-3.1
+  def process_request(request, config \\ []) do
+    # There is no resource owner in this step so we pass nil to the prehandler
+    nil
+    |> Utils.prehandle_request(request, config)
+    |> validate_request(config)
+    |> issue_grant(config)
+    |> send_response(config)
+  end
+
+  def send_response({:ok, %{grant: grant}}, _config) do
+    # TODO: Make interval configurable and verification_uri based on configured
+    # host/path
+    payload = %{
+      device_code: grant.device_code,
+      expires_in: grant.expires_in,
+      interval: 5,
+      user_code: grant.user_code,
+      verification_uri: "https://authorization-server.com/device"
+    }
+
+    {:ok, payload}
+  end
+
+  def send_response(
+        {:error, %{error: error, error_http_status: error_http_status}},
+        _config
+      ) do
+    {:error, error, error_http_status}
+  end
+
+  defp generate_grant_params(request, config) do
+    device_code = generate_device_code()
```

Comment:
Could these be in the map and not use temp assignments?

Reply:
yup!